### DISCUSSION TITLE
Protocol "curl https" not supported or disabled in libcurl

### DISCUSSION THREAD

**Discussion by fr3ak2402** [curl/curl#11545] [Q&A] [answered]
**I did this**

I have raised the version Curl 7.75 to version 8.2.0. Unfortunately, I always get in the log, an error message that describes: Protocol "curl https" not supported or disabled in libcurl".

In the project Cmake is used, so I have adapted the Curl.cmake, so that this points to the version Curl 8.2.0.

The following version of OpenSSL is used: 1.1.1k. I have also already raised this version to version 1.1.1u. Unfortunately I get the same message in the log.

The following Curl.cmake is used:

```
set(_curl_platform_flags 
  -DENABLE_IPV6:BOOL=ON
  -DENABLE_VERSIONED_SYMBOLS:BOOL=ON
  -DENABLE_THREADED_RESOLVER:BOOL=ON
  -DENABLE_MANUAL:BOOL=OFF
  -DCURL_DISABLE_LDAP:BOOL=ON
  -DCURL_DISABLE_LDAPS:BOOL=ON
  -DCURL_DISABLE_RTSP:BOOL=ON
  -DCURL_DISABLE_DICT:BOOL=ON
  -DCURL_DISABLE_TELNET:BOOL=ON
  -DCURL_DISABLE_POP3:BOOL=ON
  -DCURL_DISABLE_IMAP:BOOL=ON
  -DCURL_DISABLE_SMB:BOOL=ON
  -DCURL_DISABLE_SMTP:BOOL=ON
  -DCURL_DISABLE_GOPHER:BOOL=ON
  -DCURL_DISABLE_TFTP:BOOL=ON
  -DCURL_DISABLE_MQTT:BOOL=ON
  #-DHTTP_ONLY=ON

  -DCMAKE_USE_GSSAPI:BOOL=OFF
  -DCMAKE_USE_LIBSSH2:BOOL=OFF
  -DUSE_RTMP:BOOL=OFF
  -DUSE_NGHTTP2:BOOL=OFF
  -DUSE_MBEDTLS:BOOL=OFF
)

if (WIN32)
  set(_curl_platform_flags  ${_curl_platform_flags} -DCMAKE_USE_SCHANNEL=ON)
elseif (APPLE)
  set(_curl_platform_flags 
    
    ${_curl_platform_flags}

    #-DCMAKE_USE_SECTRANSP:BOOL=ON 
    -DCMAKE_USE_OPENSSL:BOOL=ON

    -DCURL_CA_PATH:STRING=none
  )
elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
  set(_curl_platform_flags 

    ${_curl_platform_flags}

    -DCMAKE_USE_OPENSSL:BOOL=ON

    -DCURL_CA_PATH:STRING=none
    -DCURL_CA_BUNDLE:STRING=none
    -DCURL_CA_FALLBACK:BOOL=ON
  )
endif ()

if (BUILD_SHARED_LIBS)
  set(_curl_static OFF)
else()
  set(_curl_static ON)
endif()

bambustudio_add_cmake_project(CURL
  # GIT_REPOSITORY      https://github.com/curl/curl.git
  # GIT_TAG             curl-7_75_0
  URL                 https://github.com/curl/curl/releases/download/curl-8_2_0/curl-8.2.0.zip
  URL_HASH            SHA256=3f36ff8e8eb780a9e1400309a1f813f260399baa93c2fc8ac0c389e2161c5098
  DEPENDS             ${ZLIB_PKG}
  # PATCH_COMMAND       ${GIT_EXECUTABLE} checkout -f -- . && git clean -df && 
  #                     ${GIT_EXECUTABLE} apply --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/curl-mods.patch
  CMAKE_ARGS
    -DBUILD_TESTING:BOOL=OFF
    -DBUILD_CURL_EXE:BOOL=OFF
    -DCMAKE_POSITION_INDEPENDENT_CODE=ON
    -DCURL_STATICLIB=${_curl_static}
    ${_curl_platform_flags}
)

if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
  add_dependencies(dep_CURL dep_OpenSSL)
endif ()

if (MSVC)
    add_debug_dep(dep_CURL)
endif ()
```

**operating system**

for macOS the update works with Curl 8.2.0 & OpenSSL 1.1.1k. On Windows it does not and I get the above message from CURL.

Can someone maybe help me solve this problem or tell me what I am doing wrong?

**Comment by bagder** [comment, marked as answer]
> Protocol "curl https" not supported or disabled in libcurl".

This means you literally used `curl https://...` as a URL that you pass to curl, which obviously is not a valid URL.

I believe more recent curl versions will output a different error message though.

**Reply by fr3ak2402** [reply]
I have this problem not with Curl 7.75 but with Curl 8.2.0....

I am planning to upgrade from Curl 7.75 to version 8.2.0.

The problem does not occur with Curl 7.75, but only when I use version 8.2.0.

**Reply by fr3ak2402** [reply]
So this writing is now wrong, if I understand correctly?

static const std::string VERSION_CHECK_URL = "https://api.github.com/repos/fr3ak2402/GalaxySlicer/releases";

**Reply by bagder** [reply]
> Protocol "curl https" not supported or disabled in libcurl".

I insist that you cannot get this error with libcurl 8.2.0, meaning that if you get this error you are not using libcurl 8.2.0.

Since many versions back, libcurl does not allow space in the URL scheme so the protocol can never become `curl https` now. Trying such a protocol now: `curl "curl https://example.com"` gets this error message: `curl: (3) URL rejected: Malformed input to a URL function`...

> So this writing is now wrong, if I understand correctly?

That line just shows how you assign a variable. It does not show how you make libcurl use it.

**Reply by fr3ak2402** [reply]
So this all works under CURL 7.75 without any problems.

If I change the URL and hash in my Cmake file to version CURL 8.2.0 and compile my dependencies, then I have CURL 8.2.0 and no longer CURL 7.75.

```
bambustudio_add_cmake_project(CURL
  # GIT_REPOSITORY      https://github.com/curl/curl.git
  # GIT_TAG             curl-7_75_0
  URL                 https://github.com/curl/curl/releases/download/curl-8_2_0/curl-8.2.0.zip
  URL_HASH            SHA256=3f36ff8e8eb780a9e1400309a1f813f260399baa93c2fc8ac0c389e2161c5098
  DEPENDS             ${ZLIB_PKG}
  # PATCH_COMMAND       ${GIT_EXECUTABLE} checkout -f -- . && git clean -df && 
  #                     ${GIT_EXECUTABLE} apply --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/curl-mods.patch
  CMAKE_ARGS
    -DBUILD_TESTING:BOOL=OFF
    -DBUILD_CURL_EXE:BOOL=OFF
    -DCMAKE_POSITION_INDEPENDENT_CODE=ON
    -DCURL_STATICLIB=${_curl_static}
    ${_curl_platform_flags}
)
```

And before I build my dependencies, I delete my folder where all the compiled dependencies are.

When I compile and run the program I get this message when a query is made to the api from GitHub. 

Before the compilation starts, the VS2022 compiler also tells me that it has CURL 8.2.0 in the dependencies...

Now I'm confused how I can't use Curl 8.2.0?

The other thing is that it works without problems with macOS and leads to problems with Windows...

**Reply by bagder** [reply]
> Now I'm confused how I can't use Curl 8.2.0?

Nobody can tell since you won't explain to us what you do.

**Comment by bch** [comment]
On Sun, Jul 30, 2023 at 14:29 fr3ak2402 ***@***.***> wrote:

> So this writing is now wrong, if I understand correctly?
>
> static const std::string VERSION_CHECK_URL = "
> https://api.github.com/repos/fr3ak2402/GalaxySlicer/releases";
>

What does your code look like that sets the CURLOPT_URL ?
